### PR TITLE
Add support for Open Course World level codes

### PR DESCRIPTION
**Pull request by demize**
### Checklist

- [x] Have you followed the guidelines in our Contributing document?
- [x] Have you run `eslint` on the code and resolved any errors?
- [x] Have you run `npm test` and all tests are passing?
- [x] Have you added new tests for any new functions created?
- [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [x] Have you updated CHANGELOG.md to add your changed under the `[Unreleased]` heading?

### Description

With the launch of Open Course World, there's now a new kind of level code people might want to accept, that shouldn't be considered a valid level code for the NSO MM2 servers. This PR will allow users to enable (and chatters to submit) OCW level codes.

### Benefits

This PR will allow the bot to be used in more situations.

### Potential drawbacks

Likely none.

### CODE REVIEW

**Comment by demize** [comment]
TODO: Better commenting and documentation. Update the CHANGELOG. Add tests (@liquidnya I might need your help on this one 😅).

But the functionality is there, and I'll likely push this out as a *very pre* prerelease to Docker Hub tomorrow so we can get people using it.

**Review by liquidnya — CHANGES_REQUESTED** [review]

File: src/extensions/ocw.js
Comment (demize) [reply src/extensions/ocw.js:13]: Ah, I got mixed up by the two different uses of "Strict" you've got in smm2.js... `codeStrict` looks for triplets ending in F, G, or H, but `levelCodeRegexStrict` looks for a string that is only a level code.

Updated the code to address this.

Comment (liquidnya) [reply src/extensions/ocw.js:13]: Good finding!

**Review by liquidnya — COMMENTED** [review]
The changes all look good to me.
@demize should i prepare a test case *.test.log file that you can easily modify to test this feature and make a PR into `feature/ocw-codes`?

**Comment by liquidnya** [comment]
Adding unit tests for `ocw.js` depends on this issue:  #44
And i think it is fine to just test the code by doing integration tests (chat-log tests) for now.
I wonder if we should rename `chat-logs.test.js` to just `integration.tests.js`

**Comment by demize** [comment]
> @demize should i prepare a test case \*.test.log file that you can easily modify to test this feature and make a PR into `feature/ocw-codes`?

Up to you. We could also merge this in (after I quickly go through and add some better comments—and after the Docker PR is merged, since this is branched from that) and create an issue to track the missing tests for OCW codes.

**Comment by liquidnya** [comment]
> Up to you. We could also merge this in (after I quickly go through and add some better comments—and after the Docker PR is merged, since this is branched from that) and create an issue to track the missing tests for OCW codes.
@demize I added tests and changed one implementation detail in #47.

File: src/extensions/ocw.js
Comment (liquidnya) [inline src/extensions/ocw.js:29]: `ocw-lenient` isn't quite working yet, but this is because I did not implement iterating through all matches.
The tests for that are commented out, but I think the changes can already be merged into develop regardless.
The reason why `smm2-lenient` works better is because of the last digit has to be F, G, or H requirement.

**Review by liquidnya — APPROVED** [review]
All that is left is rebasing this branch.

**Comment by demize** [comment]
I'll go through and make sure things are a little better commented now, then squash and merge into develop.

File: src/extensions/ocw.js
Comment (demize) [reply src/extensions/ocw.js:29]: Hmm... even iterating through all matches might not work here. We might want to rearchitect the lenient versions a little.

Testing on regex101, it actually only picks up two matches: `please pla` and `vel N41-Q4W`. Dropping space as a deliminator makes it work with that code, though I'm not convinced that's the best solution... I'll open an issue for this, we can discuss it more there.

**Pull request merged**
